Задания к 3-му уроку

shawtylo opened the pull request.

shawtylo committed:
Задания к 3-му уроку

shawtylo committed:
Задания к 3-му уроку

hyragano commented on src/lesson3/home/work/exercise_1/MinValueArray.java:
```diff
@@ -0,0 +1,65 @@
+
+    private static int[] getRandomChanges(int[] arrayForExercise, int[] valuesForChanges) {
+        for (int i = 0; i < valuesForChanges.length; i++) {
+            List<Integer> listForExerciseArray = new ArrayList<>(valuesForChanges.length);
```
мы пока массивы проходим, с коллекциями и врапперами пока лучше не играться

hyragano commented on src/lesson3/home/work/exercise_1/MinValueArray.java:
```diff
@@ -0,0 +1,65 @@
+            List<Integer> listForExerciseArray = new ArrayList<>(valuesForChanges.length);
+            List<Integer> listForValuesArray = new ArrayList<>(valuesForChanges.length);
+            int randomElementFromExerciseArray = new Random().ints(1, 0, arrayForExercise.length - 1).findFirst().getAsInt();
+            int randomElementFromValuesArray = new Random().ints(1, 0, valuesForChanges.length - 1).findFirst().getAsInt();
```
жуть, мб не стоит рандомы пока юзать?

hyragano commented on src/lesson3/home/work/exercise_3/MethodForTatarstan.java:
```diff
@@ -0,0 +1,60 @@
+        }
+    }
+
+    public static boolean getBinarySearch(int[] arrayForSearching, int valueForSearching) {
```
просто search  или exists, можно как-то так

shawtylo replied:
исправил

hyragano commented on src/lesson3/home/work/exercise_3/MethodForTatarstan.java:
```diff
@@ -0,0 +1,60 @@
+        return arrayForSorting;
+    }
+
+    public static int getRandomArrayElementNumber (int size, int minSize, int maxSize){
```
не стоит, тут еще есть проблема: у тебя 3 параметра с одним и тем же типом, это называется telescopic method. убираем пока что, но попытка норм

hyragano commented on src/lesson3/home/work/exercise_1/MinValueArray.java:
```diff
@@ -0,0 +1,65 @@
+// Посчитать сколько элементов массива не равны Integer.MIN_VALUE. Таким образом получим сколько элементов менялось от первоначального.
+package lesson3.home.work.exercise_1;
+import java.util.*;
+import java.util.stream.Stream;
```
ебиться сердце.... давай пока откатимся лучше,  no streams (не ожидал, что буду запрещать когда-либо стримы))))

shawtylo replied:
честно, пытался это просто через списки, но так и не победил. всё что со стримами в генераторе массива , больше заслуга ide

hyragano commented on src/lesson3/home/work/exercise_1/MinValueArray.java:
```diff
@@ -0,0 +1,65 @@
+    public static void main(String[] args) {
+        
+        int arraySizeForExercise = getRandomArraySizeForExercise(1,8,16);
+//        int arraySizeForExercise = 4;
```
За коменты буду бить по рукам

shawtylo replied:
всё почистил

shawtylo committed:
Задания к 3-му уроку - исправления

shawtylo reviewed (commented):
@hyragano 
в общем, да, пока с экспериментами надо аккуратнее. но вроде победил. правда от стримов не избавился - сделал генератор массива через shuffle, и вот приведение списков в массивы и наоборот без стримов, как то не удалось придумать решение. 
но тут больше про то,что я пока о нем и не знаю)))

shawtylo commented on src/lesson3/home/work/exercise_1/MinValueArray.java:
```diff
@@ -7,59 +7,48 @@
+
+    private static int[] getRandomChanges(int[] arrayForExercise, int[] valuesForChanges){
+        System.arraycopy(valuesForChanges, 0, arrayForExercise, 0, valuesForChanges.length);
+        List<Integer> res = Arrays.asList(Arrays.stream(arrayForExercise).boxed().toArray(Integer[]::new));
```
уже заметил. сменил название на newArrayForExercise

shawtylo committed:
Merge remote-tracking branch 'origin/dev' into dev

# Conflicts:
#	src/lesson3/home/work/exercise_1/MinValueArray.java

shawtylo committed:
Задания к 3-му уроку - доработка 1.02:
- MinValueArray - исправленное название временного list для метода
- MethodForTatarstan - добавлен метод getSearchWithWhile, идентичный методу getSearch,но с использованием цикла while
